### ISSUE TITLE
split `get_cwl_task_outputs` into separate functions for the separate uses 

### ISSUE THREAD

**Issue by mr-c** [common-workflow-lab/wdl-cwl-translator#157]
I think we are overloading `get_cwl_task_outputs` with too many tasks to accomplish by different callers of the function.

1. [`load_wdl_workflow()` line 233](https://github.com/common-workflow-lab/wdl-cwl-translator/blob/4aa8229bac7d1f42e48cae06229334398d5bb525/wdl2cwl/main.py#L233) building the CWL version of the workflow level outputs using the given the `effective_outputs()` object (`WDL.Env.Bindings[Decl]`) to get the following
- name of the workflow output (matching the WDL name of the workflow output, to be nice to the WDL users)
- CWL type for the WDL type of the output
- name of the step to get the output from
- name of the output from that step

2. [`load_wdl_workflow()` line 289](https://github.com/common-workflow-lab/wdl-cwl-translator/blob/4aa8229bac7d1f42e48cae06229334398d5bb525/wdl2cwl/main.py#L289) building the `cwl.WorkflowStepOutput` object using the current step (either a `WDL.Tree.Task` or `WDL.Tree.Workflow`)'s outputs (`List[WDL.Tree.Decl]`) to get the names (`str`) of the outputs

3. [`load_wdl_task()`](https://github.com/common-workflow-lab/wdl-cwl-translator/blob/4aa8229bac7d1f42e48cae06229334398d5bb525/wdl2cwl/main.py#L315) the nominal user of `get_cwl_task_outputs`; uses the entire list of `cwl.CommandOutputParameter` objects as-is to build the `cwl.CommandLineTool` from the `WDL.Tree.Task` outputs (`List[WDL.Tree.Decl]`)

In summary: three users of the same function, with three different sets of goals, and three different inputs :-)

I recommend we create three separate functions for these different purposes; there could be a helper function used by at least two of the new functions.

What do you think @Th3nn3ss @kinow ?

**Comment by kinow** [comment]
+1 ! Good spotting.

@Th3nn3ss there's a concept that can be useful for understanding why what @mr-c is suggesting here is good, and that's really common in programming job interviews: [SOLID](https://en.wikipedia.org/wiki/SOLID)

The `S` in SOLID stands for [single-responsibility](https://en.wikipedia.org/wiki/Single-responsibility_principle).

p.s. every rule has exceptions, so it doesn't mean you need to apply SOLID everywhere, to every class and function, etc… time & experience will give you a feeling of when these sort of rules can/must be applied :+1:
